File: LeetCode/908-middle-of-the-linked-list/middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode * temp= head;
        int count =0;
        while(temp!=NULL){
            count++;
            temp=temp->next;
        }
        temp=head;
        for(int i=0; i< (count/2);++i){
            temp=temp->next;
        }
        return temp;
        
    }
};

File: LeetCode/908-middle-of-the-linked-list/middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    var slow = head;
    var fast = head;
    while(fast != null && fast.next != null){
        slow = slow.next;
        fast= fast.next;
        fast= fast.next;

    }
    return slow;
};